### PR TITLE
MBS-7473: Allow specifying a release MBID when attaching a CD TOC

### PR DESCRIPTION
**Pull request by mwiencek**
If the specified MBID can't have the CD TOC attached, I figured it'd be confusing to show an empty results page, so it shows the requested release with a custom error message.

### CODE REVIEW

File: lib/MusicBrainz/Server/Controller/CDTOC.pm
Comment (reosarevok) [inline lib/MusicBrainz/Server/Controller/CDTOC.pm]: Ideally this would check if it contains a GUID like most of our other searches, so you can still paste a release link directly, but this is certainly an improvement.

File: lib/MusicBrainz/Server/Controller/CDTOC.pm
Comment (reosarevok) [inline lib/MusicBrainz/Server/Controller/CDTOC.pm]: I would prefer returning something that specifies that it understood it's an MBID, but it didn't match anything - rather than the standard "No results found. Try refining your search query." it could be something like "We didn't find a release with that MBID" or "That MBID doesn't seem to belong to a release". Not sure what the best way of achieving that would be - I guess you could set a boolean for $was_mbid_search and pass it to the stash?

File: root/cdtoc/attach_list.tt
Comment (reosarevok) [inline root/cdtoc/attach_list.tt]: This doesn't actually check for wrong format though. If you pick an MBID for a release with the wrong format it will load it without the error, it'll just be unselectable:

Screenshot from 2020-02-22 19-32-51

File: lib/MusicBrainz/Server/Controller/CDTOC.pm
Comment (mwiencek) [reply lib/MusicBrainz/Server/Controller/CDTOC.pm]: Changed it to use a regular expression to extract the MBID, which'll allow people to paste a link.

File: lib/MusicBrainz/Server/Controller/CDTOC.pm
Comment (mwiencek) [reply lib/MusicBrainz/Server/Controller/CDTOC.pm]: Was hoping I could reuse the message from root/mbid/NotFound.js, but seems a bit too long and generic. Opted for "We couldn’t find a release matching that MBID." Let me know if that can be improved.

File: root/cdtoc/attach_list.tt
Comment (mwiencek) [reply root/cdtoc/attach_list.tt]: I changed the message to not mention formats, and also only show this when it's an MBID search now.

**Comment by mwiencek** [comment]
@brainzbot retest this please

**Review by reosarevok — APPROVED** [review]
Seems to work fine now.

File: root/cdtoc/attach_filter_release.tt
Comment (yvanzo) [inline root/cdtoc/attach_filter_release.tt:9]: Since it’s a new string:

```suggestion
      [% r.label('query', add_colon(l('Release title or MBID'))) %]
```

File: root/cdtoc/lookup.tt
Comment (yvanzo) [inline root/cdtoc/lookup.tt:71]: Same as above:

```suggestion
    [% form_row_text(r, 'query', add_colon(l('Release title or MBID'))) %]
```

**Review by yvanzo — APPROVED** [review]
LGTM but new localizable message, shouldn’t require a second review though.

**Pull request merged**
